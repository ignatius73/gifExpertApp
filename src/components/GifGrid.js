import React, { useEffect } from 'react';

export const GifGrid = ({category}) => {

    useEffect( () => {
        getGiphes();
    }, []);

    const getGiphes = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q={category}&limit=10&api_key=B4VMClSog6hZwFQ5i6KsnU7CgUxUWeDJ'
        const resp = await fetch(url);
        const { data } = await resp.json();
        
        const gifs = data.map( img => {
            return{
                id: img.id,
                gif: img.images?.downsized_large.url,
                title: img.title   
            }
        });
            
        console.log(gifs);
        
    }

    getGiphes();
    return (
                <>
                    <h3>{category}</h3>
                </>
            )
}
