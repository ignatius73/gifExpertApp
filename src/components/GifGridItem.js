import React from 'react'

export const GifGridItem = (props) => {

    console.log(props);

    return (
        <div className='card animate__animated animate__bounce animate__jackInTheBox'>
          {  <img src={props.gif} alt={props.title} /> }
        </div>
    )
}
