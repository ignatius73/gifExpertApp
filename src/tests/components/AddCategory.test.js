import React from 'react';
import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"

describe('Unitary tests on AddCategory', () => {

    const setcategories = jest.fn();
    let wrapper;

    //Reinicializo el wrapper por cada test
    beforeEach( ()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setcategories } />);
    })

    test('should get a Category', () => {
       
        expect( wrapper ).toMatchSnapshot();

    })

    test('should InputChange', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', {target: { value} });

        expect( wrapper.find('p').text().trim()).toBe(value);
        
        
    });

    test('shouldnt send submit button', () => {
      wrapper.find('form').simulate('submit', { preventDefault(){} });  
      expect( setcategories ).not.toHaveBeenCalled();
    
    })
    
    
    
    
})
