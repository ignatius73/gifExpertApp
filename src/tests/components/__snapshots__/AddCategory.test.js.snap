// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unitary tests on AddCategory should get a Category 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
